app/src/org/commcare/activities/FormEntryActivityUIController.java

```diff
@@ -759,7 +759,9 @@ protected void updateFormRelevancies() {
             if (oldWidgets.get(i) instanceof ImageWidget) {
                 // If there was change(in particular image-remove) in an image widget,
                 // only then update the form to disk.
-                activity.onExternalAttachmentUpdated();
+                if (i == indexOfLastChangedWidget) {
```

Comment:
we should put this in the outer `if` itself as the first condition so that we only do the `instanceof ImageWidget` check when required.